docs/DocumentOutlineAndTableOfContents.md

```diff
@@ -29,21 +29,21 @@ Note that a page break will always be triggered after inserting the table of con
 
 ## With HTML ##
 
-When using [`fpdf.HTMLMixin`](HTML.md), a document outline is automatically built.
+When using [`FPDF.write_html`](HTML.md), a document outline is automatically built.
 You can insert a table of content with the special `<toc>` tag.
 
 Custom styling of the table of contents can be achieved by overriding the `render_toc` method
 in a subclass of `fpdf.html.HTML2FPDF`:
 ```python
-from fpdf import FPDF, HTMLMixin, HTML2FPDF
+from fpdf import FPDF, HTML2FPDF
 
 class CustomHTML2FPDF(HTML2FPDF):
     def render_toc(self, pdf, outline):
         pdf.cell(txt='Table of contents:', new_x="LMARGIN", new_y="NEXT")
         for section in outline:
             pdf.cell(txt=f'* {section.name} (page {section.page_number})', new_x="LMARGIN", new_y="NEXT")
 
-class PDF(FPDF, HTMLMixin):
+class PDF(FPDF):
     HTML2FPDF_CLASS = CustomHTML2FPDF
```

Comment:
Actually, I think we should not recommend using `HTML2FPDF_CLASS` anymore.
We just preserve it for backward compatibility, but remove it from the docs here.
A simpler solution now is just to override the `render_toc()` method in a subclass of `FPDF`.
Could you please make that change here and in `test_custom_HTML2FPDF`?

Reply:
update the usage at 78c342fe6b2e4fe1923c62d11a813acdd490b4a9

Reply:
The test case failed. let me recheck the code.

Reply:
I wasn't able to resolve the issue so I want to ask you a question.

I have a question.  `HTML2FPDF_CLASS`  is currently instantiated in the `write_html` method and I think `render_toc()` method can't be overridden in the `HTML2FPDF` class except that we overwrite HTML2FPDF_CLASS as the way of using `HTML2FPDF_CLASS`.

Sorry, I can't come up with a solution.

```python

 def write_html(self, text, *args, **kwargs):
        """Parse HTML and convert it to PDF"""
        kwargs2 = vars(self)
        # Method arguments must override class & instance attributes:
        kwargs2.update(kwargs)
        h2p = self.HTML2FPDF_CLASS(self, *args, **kwargs2)
        if self.HTML2FPDF_CLASS != HTML2FPDF:
            warnings.warn(
                "The HTML2FPDF_CLASS is deprecated. It will be removed in a future release.",
                DeprecationWarning,
                stacklevel=2,
            )
        text = html.unescape(text)  # To deal with HTML entities
        h2p.feed(text)
```

Reply:
I *think* you can remove the warning here alltogether.

We only want to warn the user about using the `HTMLMixin` class, which now issues that warning in its `__init__()` method.

Reply:
> I have a question. `HTML2FPDF_CLASS` is currently instantiated in the `write_html` method and I think `render_toc()` method can't be overridden in the `HTML2FPDF` class except that we overwrite HTML2FPDF_CLASS as the way of using `HTML2FPDF_CLASS`.

You are right, I was mistaken, there is no easy way around this.
Let's keep it as it is for now, your PR is already great as it is!

Reply:
I restored the test and sample code when `HTML2FPDF_CLASS` was used in `test_outline_html.py` and `DocumentOutlineAndTableOfContents.md`.